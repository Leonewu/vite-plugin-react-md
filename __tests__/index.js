const plugin = require('../lib/index').default

const { transform } = plugin()

it('basic', () => {
  const result = transform(`
  # title
  - a
  - b
  ## title2
  1. a
  2. b
  `, 'test.md')
  expect(result.code).toMatchSnapshot()
});

it('code fence', () => {
  const result = transform(`
  # title
  ## title2
  \`test\`
  \`\`\`js
  { 
    a: 123
  }
  \`\`\`
  `, 'test.md')
  expect(result.code).toMatchSnapshot()
});