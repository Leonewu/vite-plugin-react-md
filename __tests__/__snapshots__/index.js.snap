// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"import React from \\"react\\";
const attributes = {};
const __html = \`<h1>title</h1>
<ul>
<li>a</li>
<li>b</li>
</ul>
<h2>title2</h2>
<ol>
<li>a</li>
<li>b</li>
</ol>
\`;
function ReactComponent(props) {
  return /* @__PURE__ */ React.createElement(\\"div\\", {
    className: \\"markdown\\",
    dangerouslySetInnerHTML: { __html }
  });
}
export {
  attributes,
  ReactComponent
};
"
`;

exports[`code fence 1`] = `
"import React from \\"react\\";
const attributes = {};
const __html = \`<h1>title</h1>
<h2>title2</h2>
<p><code>test</code></p>
<pre><code class=\\"language-js\\">{ 
  a: 123
}
</code></pre>
\`;
function ReactComponent(props) {
  return /* @__PURE__ */ React.createElement(\\"div\\", {
    className: \\"markdown\\",
    dangerouslySetInnerHTML: { __html }
  });
}
export {
  attributes,
  ReactComponent
};
"
`;
